server_vllm.py

```diff
@@ -227,11 +237,46 @@ async def create_chat_completion(raw_request: Request):
             HTTPStatus.BAD_REQUEST, "logit_bias is not currently supported"
         )
 
+    if request.functions:
+        tools_or_functions = [item.dict() for item in request.functions]
+    elif request.tools:
+        tools_or_functions = [item.dict() for item in request.tools]
+    else:
+        tools_or_functions = []
+
+    # Adjust tools_or_functions based on tool_choice
```

Comment:
This logic is different from inference.py ? in Inference.py, we didn't adjust tools_or_functions

Reply:
I think we should implement a function in inference.py for adjusting tools_or_functions based on tool_choice and call this in both inference and server_vllm.py

Reply:
In inference.py, we shouldn't need to care about adjusting `tools_or_functions` since it is used by both training and inference. We only need to check whether to add any forced function to the end of the prompt. Thus, I feel that the adjusting part should only be in server_vllm.py.

Reply:
Oh the reason I think we should fix in inference.py instead of server_vllm.py because inference.py is used by both server.py and server_vllm.py so we won't duplicate the code ?